Make it possible to exclude directories from rebuild-pot

gyst opened the pull request:
Previously, only filenames like `ignoreme.html` or globs like `ignoreme.*` could be used in `--exclude=...`.
This PR makes it possible to use `--exclude="directoryname"` in addition to the file exclusions.

ale-rt reviewed (commented):
The biggest concern I have is that this is not os independent.
The code might already not work on windows (I never tested it) but `os.path.sep` instead of assuming that the separator is `/` seems a better choice to me.
In addition it seems the code works in a unexpected way... 
Passing `bar` or `bar/` will make an enormous difference for no reason if you have a directory called `foobar`.
Maybe adding some tests will help exposing other code weaknesses.

ale-rt commented on src/i18ndude/extract.py:
```diff
@@ -436,6 +436,15 @@ def find_files(dir, pattern, exclude=()):
+        # exclude dirnames
+        for ex in exclude:
+            # exact dirname match
+            if ex in dirpath.split("/"):
```
Generally speaking `dirpath.split(os.path.sep)` will repeat equally in every cycle so a minor improvement would be moving this outside the for block

gyst commented:
I have addressed all your comments, except for the tests: there is no test coverage on `rebuild-pot` *at all* currently so I do not have a baseline I can expand.

ale-rt reviewed (approved):
Thanks!

mauritsvanrees commented:
Strange: the GitHub Actions that I added earlier this year were not run for this PR. Here is the [config](https://github.com/collective/i18ndude/blob/master/.github/workflows/main.yml). On the [Actions tab](https://github.com/collective/i18ndude/actions) I did see a warning that the scheduled tests were disabled due to inactivity. But that should only have an effect on the monthly tests started with cron, I would hope. I enabled them again.

Problem: the tests fail locally. And I have pushed a commit to master to trigger a new run, and the failures can be seen [there as well](https://github.com/collective/i18ndude/runs/2693701425?check_suite_focus=true).

When I go back to commit cf6bfb1c5bdb192e08cff3004e20d565676b6973 which merged your other PR #85, the tests pass. So the current PR breaks something.  Can you have a look?  I don't immediately see a relation to what you changed.

gyst commented:
OK I'll have a look at those fails, thanks.
